File: problem_5/main.js
#!/usr/bin/env node

function cons(a, b) {
    return function pair(f) {
        return f(a, b);
    };
}

/////////////// ignore above this line ////////////////////
function car(cons) {
    return cons(function(a) {
        return a;
    });
}

function cdr(cons) {
    return cons(function(a, b) {
        return b;
    });
}

function main() {
    console.log('car: ' + car(cons(3, 4)));
    console.log('cdr: ' + cdr(cons(3, 4)));
}

main();

File: problem_5/main.py
#!/usr/bin/env python3

def cons(a, b):
    def pair(f):
        return f(a, b)
    return pair

#/////////////// ignore above this line ////////////////////
def car(cons):
	def fnc(a, b):
		return a
	return cons(fnc)

def cdr(cons):
	def fnc(a, b):
		return b
	return cons(fnc)

fnc = cons(3, 4)
print('car: ', car(fnc))
print('cdr: ', cdr(fnc))
